### PR TITLE
feat(cryptocom): add ws trading api

### PR DESCRIPTION
**Pull request by pcriadoperez**
- Add trading api for ws `createOrderWs`, `cancelOrderWs`, cancelAllOrdersWs`
- Fix error handling in python ws
- Cryptocom allows all it's rest endpoints to also work by websockets. Can continue adding other endpoints in following PRs

### CODE REVIEW

**Comment by pcriadoperez** [comment]
```python
cryptocom.createOrderWs(BTC/USDT,market,buy,0.0001)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_fd3e9d2f8e5d01ef92a0e1',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895314350602',
 'info': {'client_oid': 'CCXT_fd3e9d2f8e5d01ef92a0e1',
          'order_id': '6142909895314350602'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
 ```

File: ts/src/pro/cryptocom.ts [hunk 1]
```diff
@@ -447,6 +451,97 @@ export default class cryptocom extends cryptocomRest {
...
+         * @method
+         * @name cryptocom#cancelOrder
+         * @description cancels an open order
+         * @see https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-cancel-order
+         * @param {string} id the order id of the order to cancel
+         * @param {string} [symbol] unified symbol of the market the order was made in
+         * @param {object} [params] extra parameters specific to the cryptocom api endpoint
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
```
Comment (carlosmiei) [inline ts/src/pro/cryptocom.ts:511]: any clue why I'm having this error?
```
 p cryptocom cancelOrderWs  "6142909895323193963"
Python v3.10.9
CCXT v4.0.10
cryptocom.cancelOrderWs(6142909895323193963)
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 239, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 470, in cancel_order_ws
    return await self.watch_private_request(messageHash, request)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 513, in watch_private_request
    return await self.watch(url, str(nonce), message, True)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 542, in handle_error_message
    self.throw_exactly_matched_exception(self.exceptions['exact'], errorCode, feedback)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 2181, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.BadRequest: cryptocom {"id":1688726256929,"method":"private/cancel-order","code":10004,"message":"BAD_REQUEST","original":"{\"id\":1688726256929,\"method\":\"private/cancel-order\",\"params\":{\"order_id\":6142909895323193963},\"nonce\":1688726256929}"}
```

File: ts/src/pro/cryptocom.ts [hunk 2]
```diff
@@ -447,6 +451,97 @@ export default class cryptocom extends cryptocomRest {
...
+        }, params);
+        const request = {
+            'method': 'private/cancel-order',
+            'params': params,
+        };
+        const messageHash = this.nonce ();
+        return await this.watchPrivateRequest (messageHash, request);
+    }
+
+    async cancelAllOrdersWs (symbol: string = undefined, params = {}) {
```
Comment (carlosmiei) [inline ts/src/pro/cryptocom.ts:523]: same here
```
p cryptocom cancelAllOrdersWs                            
Python v3.10.9
CCXT v4.0.10
cryptocom.cancelAllOrdersWs()
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 239, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 490, in cancel_all_orders_ws
    return await self.watch_private_request(messageHash, request)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 513, in watch_private_request
    return await self.watch(url, str(nonce), message, True)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 542, in handle_error_message
    self.throw_exactly_matched_exception(self.exceptions['exact'], errorCode, feedback)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 2181, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.BadRequest: cryptocom {"id":1688726363275,"method":"private/cancel-all-orders","code":10004,"message":"BAD_REQUEST","original":"{\"id\":1688726363275,\"method\":\"private/cancel-all-orders\",\"params\":{},\"nonce\":1688726363275}"}
```

File: ts/src/pro/cryptocom.ts (hunk 1 above)
Comment (pcriadoperez) [reply ts/src/pro/cryptocom.ts:534]: It was using the old exchange url instead of the unified url, I merged main with Dan's latest PR and should now work.

**Comment by pcriadoperez** [comment]
### cancelAllOrdersWs
```python
cryptocom.cancelAllOrdersWs()
{'code': 0,
 'id': 1688915752052,
 'message': 'NO_ERROR',
 'method': 'private/cancel-all-orders'}
 ```

**Comment by pcriadoperez** [comment]
### cancelOrder
```python
cryptocom.cancelOrderWs(6142909895352282903)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_880b6113b451c35d90b226',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895352282903',
 'info': {'client_oid': 'CCXT_880b6113b451c35d90b226',
          'order_id': '6142909895352282903'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
 ```

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
 p cryptocom createOrder "LTC/USD:USD" limit buy 0.1 50 
Python v3.10.9
CCXT v4.0.10
cryptocom.createOrder(LTC/USD:USD,limit,buy,0.1,50)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_fc6aff78e5950d6b21ef43',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895323193963',
 'info': {'client_oid': 'CCXT_fc6aff78e5950d6b21ef43',
          'order_id': '6142909895323193963'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': 'LTC/USD:USD',
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
```

```
 p cryptocom createOrderWs "LTC/USD:USD" limit buy 0.1 50 
Python v3.10.9
CCXT v4.0.10
cryptocom.createOrderWs(LTC/USD:USD,limit,buy,0.1,50)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_43a2db1b5083f726e6719e',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895323293746',
 'info': {'client_oid': 'CCXT_43a2db1b5083f726e6719e',
          'order_id': '6142909895323293746'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
```

```
p cryptocom cancelAllOrdersWs                                                     
Python v3.10.9
CCXT v4.0.14
cryptocom.cancelAllOrdersWs()
{'code': 0,
 'id': 1688917606695,
 'message': 'NO_ERROR',
 'method': 'private/cancel-all-orders'}
```

```
 p cryptocom createOrderWs "LTC/USDT" market buy 0.1 
Python v3.10.9
CCXT v4.0.14
cryptocom.createOrderWs(LTC/USDT,market,buy,0.1)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_2ca6bd89f852c45f6945da',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895352545763',
 'info': {'client_oid': 'CCXT_2ca6bd89f852c45f6945da',
          'order_id': '6142909895352545763'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
```

```
 p cryptocom createOrderWs "LTC/USDT" limit buy 0.1 50
Python v3.10.9
CCXT v4.0.14
cryptocom.createOrderWs(LTC/USDT,limit,buy,0.1,50)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_81173cf1a79fa390700f14',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895352552921',
 'info': {'client_oid': 'CCXT_81173cf1a79fa390700f14',
          'order_id': '6142909895352552921'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
 
```

```
p cryptocom cancelOrderWs "6142909895352552921"      
Python v3.10.9
CCXT v4.0.14
cryptocom.cancelOrderWs(6142909895352552921)
{'amount': None,
 'average': None,
 'clientOrderId': 'CCXT_81173cf1a79fa390700f14',
 'cost': None,
 'datetime': None,
 'fee': {'cost': None, 'currency': None},
 'fees': [{'cost': None, 'currency': None}],
 'filled': None,
 'id': '6142909895352552921',
 'info': {'client_oid': 'CCXT_81173cf1a79fa390700f14',
          'order_id': '6142909895352552921'},
 'lastTradeTimestamp': None,
 'lastUpdateTimestamp': None,
 'postOnly': None,
 'price': None,
 'reduceOnly': None,
 'remaining': None,
 'side': None,
 'status': None,
 'stopLossPrice': None,
 'stopPrice': None,
 'symbol': None,
 'takeProfitPrice': None,
 'timeInForce': None,
 'timestamp': None,
 'trades': [],
 'triggerPrice': None,
 'type': None}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**